dds_cli/project_status.py

```diff
@@ -148,6 +149,117 @@ def update_status(self, new_status, deadline=None, is_aborted=False, no_mail=Fal
+        )
+
+        # Structure of the response:
+        #   {
+        #   'default_unit_days': 30,
+        #   'project_info': {
+        #       'Created by': 'First Unit User',
+        #       'Description': 'This is a test project',
+        #       'Last updated': 'Wed, 18 Oct 2023 08:40:43 GMT',
+        #       'PI': 'support@example.com',
+        #       'Project ID': 'project_1',
+        #       'Size': 0,
+        #       'Status': 'Available',
+        #       'Title': 'First Project'
+        #       },
+        #   'project_status': {
+        #       'current_deadline': 'Sat, 04 Nov 2023 23:59:59 GMT',
+        #       'current_status': 'Available'},
+        #   }
+
+        # Check that the returned information was ok
+        keys = ["project_info", "project_status", "default_unit_days"]
+        (
+            project_info,
+            project_status,
+            default_unit_days,
+            *_,
+        ) = dds_cli.utils.get_required_in_response(keys=keys, response=response_json)
+
+        # Check and extract the required information for the operation
+        current_status, *_ = dds_cli.utils.get_required_in_response(
+            keys=["current_status"], response=project_status
+        )
+
+        # if the project is still in progress it won't have a current_deadline parameter
+        if not current_status == "Available":
+            raise exceptions.DDSCLIException(
+                "You can only extend the deadline for a project that has the status 'Available'."
+            )
+
+        current_deadline, *_ = dds_cli.utils.get_required_in_response(
+            keys=["current_deadline"], response=project_status
+        )
+        project_id, *_ = dds_cli.utils.get_required_in_response(
+            keys=["Project ID"], response=project_info
+        )
+
+        # print information about the project status and table with the project info
+        print_info = (
+            f"\nCurrent deadline: [b][green]{current_deadline}[/green][/b]\n"
+            f"Default deadline extension: [b][green]{default_unit_days}[/green][/b] days\n"
+        )
+        table = self.generate_project_table(project_info=project_info)
+        dds_cli.utils.console.print(table)
+        dds_cli.utils.console.print(print_info)
+
+        # If it wasnt provided during the command click, ask the user for the new deadline
+        if not new_deadline:
+            # Question number of days to extend the deadline
+            prompt_question = (
+                "How many days would you like to extend the project deadline with? "
+                "Leave empty in order to choose the default"
+            )
+            new_deadline = rich.prompt.IntPrompt.ask(prompt_question, default=default_unit_days)
+
+        # Confirm operation question
+        new_deadline_date = parse(current_deadline) + datetime.timedelta(days=new_deadline)
+        new_deadline_date = new_deadline_date.strftime("%d/%m/%Y, %H:%M:%S")
```

Comment:
If not too much hassle it'd be nice to have the deadlines in the same format. Currently we have this:
```bash
Current deadline: Mon, 22 Jan 2024 23:59:59 GMT
Default deadline extension: 90 days

How many days would you like to extend the project deadline with? 
Leave empty in order to choose the default (90): 2

The new deadline for project someunit00002 will be: 24/01/2024, 
23:59:59
```

But it'd be nice to have this
```bash
Current deadline: Mon, 22 Jan 2024 23:59:59 GMT
Default deadline extension: 90 days

How many days would you like to extend the project deadline with? 
Leave empty in order to choose the default (90): 2

The new deadline for project someunit00002 will be: Wed, 24 Jan 2024 23:59:59
```

Reply:
done!